config/routes.rb

```diff
@@ -150,6 +150,12 @@
       get 'invitations' => 'users#invitations', as: :users_invitations
 
       resources :groups, except: [:show]
+
+      namespace :material, path: 'materials' do
+        resources :folders, except: [:new, :index] do
+          get 'new_subfolder', on: :member
```

Comment:
Maybe we should do /courses/X/materials/folder/Y/new/subfolder and FX-9750G /courses/X/materials/folder/Y/new/file?